hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/key/acl/OMKeyAclResponse.java

```diff
@@ -36,27 +34,30 @@
  */
 public class OMKeyAclResponse extends OMClientResponse {
 
-  private final OmKeyInfo omKeyInfo;
+  private OmKeyInfo omKeyInfo;
 
-  public OMKeyAclResponse(@Nullable OmKeyInfo omKeyInfo,
-      @Nonnull OMResponse omResponse) {
+  public OMKeyAclResponse(@Nonnull OMResponse omResponse,
+      @Nonnull OmKeyInfo omKeyInfo) {
     super(omResponse);
     this.omKeyInfo = omKeyInfo;
   }
 
+  /**
+   * For when the request is not successful or it is a replay transaction.
+   * For a successful request, the other constructor should be used.
+   */
+  public OMKeyAclResponse(@Nonnull OMResponse omResponse) {
+    super(omResponse);
+  }
+
   @Override
   public void addToDBBatch(OMMetadataManager omMetadataManager,
       BatchOperation batchOperation) throws IOException {
 
-    // If response status is OK and success is true, add to DB batch.
-    if (getOMResponse().getStatus() == OzoneManagerProtocolProtos.Status.OK &&
```

Comment:
Was this check moved somewhere else?

Reply:
Yes Arpit. OmClientResponse#checkAndUpdateDB() performs this check before calling addToDBBatch(). OMDoubleBuffer now calls checkAndUpdateDB() instead of addToDBBatch() directly.